### DISCUSSION TITLE
Changing query string names with an associative array

### DISCUSSION THREAD

**Discussion by dominiced** [livewire/livewire#1905] [3. Feature ideas]
Hello.

I really like the $validationAttributes property which you can use in Livewire 2.3.0. It's really handy to be able to provide a custom error message name for an attribute.

I think we should do the same thing for query string parameters, example below...
```
protected $queryString = [
        'sortField' => ['except' => 'id'],
    ];

    protected $queryStringNames = [
        'sortField' => 'sort',
    ];

    /**
     * This property determines which field we'll be sorting by.
     *
     * @var string
     */
    public $sortField = 'id';
```

The new property is $queryStringNames, which could be a livewire property which sets the query string key to 'sort' instead of 'sortField' and Livewire could handle translating what property that's actually pointing too.

The example is trivial, but the real reason I would like this feature is because I'm using multiple instances of the same livewire component in my page, and it's causing conflicts. If I was able to prepend/append an identifier to the query strings for each, it would resolve that.

This would also resolve the issue of withPagination conflicts. If you user were to use an $appendix property to make 'page' equals something like 'page1' and 'page2' for the first and second table respectively, the query string ?page1=&page2= wouldn't clash and Livewire would know because one component would have 'page1' in it's $queryStringNames array and likewise with the second table.

**Comment by calebporzio** [comment]
I could see the desire for something like this:

```php
public $foo = '';

protected $queryString = ['foo' => ['except' => '', 'as' => 'bar']];
// Now the query string will show "bar" instead of "foo" for the value name.
```

**Comment by dominiced** [comment]
Good thinking, that's a lot neater than my suggestion, and ties in nicely with the existing QueryString options!
